### PR TITLE
Implement the unit tests for managedfleetnotification and managednotificationrecord apis

### PR DESCRIPTION
**Pull request by shitaljante**
### What type of PR is this?
test

### What this PR does / why we need it?

- Added test cases to improve coverage.
- Removed dead code function: **NewNotificationRecord**

### Which Jira/Github issue(s) this PR fixes?

Fixes : [OSD-15520](https://issues.redhat.com/browse/OSD-15520)

### Pre-checks (if applicable):
- [ ] Ran `go test` command
- [ ] Ran `make generate` command locally to validate code changes

### CODE REVIEW

File: api/v1alpha1/managednotification_types.go [hunk 1]
```diff
@@ -245,15 +245,6 @@ func (conditions Conditions) GetCondition(t NotificationConditionType) *Notifica
 	return nil
 }
 
-// NewNotificationRecord adds a new notification record status for the given name
```
Comment (mrbarge) [inline api/v1alpha1/managednotification_types.go:248]: I've doublechecked that this is safe, which it is, but in future I'd recommend in general that we leave this sort of API change to either a separate PR or at least a separate commit, as it's not really related to the intent of the PR/ticket.

**Comment by mrbarge** [comment]
lgtm in general, can you please squash the commits? Try the following label (if it doesn't work for me):
 /label tide/merge-method-squash

File: api/v1alpha1/managednotification_types.go (hunk 1 above)
Comment (shitaljante) [reply api/v1alpha1/managednotification_types.go:248]: Ack, I agree that new PR/ticket will be good from tracking perspective. 
Thanks.

**Pull request merged**